### DISCUSSION TITLE
NoMachine (Installer hash does not match)

### DISCUSSION THREAD

**Discussion by jamaya77** [microsoft/winget-pkgs#109009] [Q&A] [answered]
Hi,

I've been trying to install "No Machine" (ID:  NoMachine.NoMachine) on my Windows 11 desktop, but it generates an error:
 
"Installer hash does not match; this cannot be overridden when running as admin"

Doesn't matter if I use an admin account or not. All other packages such "NotePad +++" and "No Machine Enterprise Client" work just fine. Any suggestions?

**Comment by mdanish-kh** [comment]
A hash mismatch generally indicates that the package has been updated / newer version has been released and the manifest needs to be updated in the repo. This will be resolved once this PR gets merged:
- https://github.com/microsoft/winget-pkgs/pull/108891

**Reply by jamaya77** [reply]
Hi,
Thanks for the quick response. Just a follow-up question, does it make any difference that the current version I'm seeing is slightly older than that one suggested in that article you posted?

 Name                        Id                        Version Match              Source
---------------------------------------------------------------------------------------
NoMachine                   NoMachine.NoMachine       8.2.3                      winget

**Reply by mdanish-kh** [reply]
Yes that is exactly why you're seeing the hash mismatch. The current _latest_ version present in the [repository is `8.2.3`](https://github.com/microsoft/winget-pkgs/tree/master/manifests/n/NoMachine/NoMachine/8.2.3) that you're seeing in the output. The Installer URL no longer works for that version resulting in a hash mismatch and a PR (linked above) needs to be created to add the newer version in the repo. Once that's merged in the repo, you'll see the newer version in the output and will be able to install the package without hash mismatches as long as its URL is valid.

**Reply by jamaya77** [reply]
Thank you. I'm a newbie and just started using winget for deployment and installing packages, which I think has been working out great rather than manually downloading the installer for different versions. One last question, I'm assuming these hash mismatch issues are common, how long after typically are these issues resolved or the URL updated?

**Reply by mdanish-kh** [reply, marked as answer]
Well for most packages that use a vanity URL i.e., a URL that always points to a latest release of a version, a hash mismatch is caught by [wingetbot](https://github.com/wingetbot), a bot that scans the InstallerUrls daily for any hash mismatches and tries to update the package with the updated hash and version. You can see the automated bots PRs [here](https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+author%3Awingetbot+is%3Aclosed). 

For other packages though including `NoMachine` that use versioned URLs i.e., the URL is unique for every new version, wingetbot doesn't have a mechanism to predict the newer version and the Installer URL. These packages are often manually updated by community contributors and users of the package that notice a newer version is out.

If you notice a hash mismatch / want a newer version in the repo, you are encouraged to create a manifest for it if you wish by [following the Contributor's Guide](https://github.com/microsoft/winget-pkgs/blob/master/CONTRIBUTING.md) or if you want someone else from the community to update it, a [package update request issue](https://github.com/microsoft/winget-pkgs/issues/new/choose) is how we generally track these mismatches

**Reply by jamaya77** [reply]
Thank you, Muhammed, for the information.

**Reply by jamaya77** [reply]
All appears to be working. Thank you.

Found NoMachine [NoMachine.NoMachine] Version 8.5.3 
Successfully verified installer hash
Starting package install...
Successfully installed.
